pycbc/inference/models/marginalized_gaussian_noise.py

```diff
@@ -241,6 +243,15 @@ def __init__(self, variable_params,
 
         self.dets = {}
 
+        if sample_rate is not None:
+            logging.info("Using %s sample rate for marginalization",
+                         sample_rate)
+            for det in self._whitened_data:
+                tlen = int(round(float(sample_rate) *
+                           self.whitened_data[det].duration))
+                flen = tlen // 2 + 1
+                self._whitened_data[det].resize(flen)
```

Comment:
Is it worth raising an error here if the sample rate is smaller than the data?

Also, this might lead to a bug... the whitened data is set when the psd is set (see [here](https://github.com/gwastro/pycbc/blob/5a9d151dd9d013c1bb651770688f589eb84f8965/pycbc/inference/models/gaussian_noise.py#L263)). In principle you can change the psd after the class is initialized. I don't think it's done in an inference run anywhere currently, but could be done in a notebook and/or possibly implemented later. I think it might be better to do this by redefining `psd.setter` so that it calls the parent, then runs this block of code to update the whitened data. You'll need to save `sample_rate` as an attribute in that case. Maybe call it something else to distinguish it from the data sample rate (`likelihood_sample_rate`?).

Reply:
Ok, interesting. Really, I just need this done before we do the FFT in the likelihood to calculate the SNR time series. I'm not sure this make more sense in the psd setter, but maybe I should put it directly in the likelihood? Is the psd setter preferable over that?

Yeah, truncating should probably be an error.

Reply:
If you're modifying the `_whitened_data` data than it would make more sense to do it in the PSD setter, to avoid the error I mentioned. But yeah, that is a round about way to just increase the sample rate of the likelihood time series. Don't you also need to zero pad the waveform in that case too? In which case, there's a bunch of 0*0 going on in the match function. What if you add this as an option to the `matched_filter_core` instead? That is, it takes in an optional sample rate argument, and if provided you zero pad `qtilde` just prior to doing the `ifft`? In the model you would store the sample rate as an attribute and pass it to `matched_filter_core` in the likelihood call.

Reply:
@cdcapano Indeed. Actually the waveform is already zeropadding to whatever the whitened data is. I guess I could just do both there (probably not too slow to just copy out the data too, so it's not replacing the original values).

Reply:
@cdcapano I'll make this change.

Reply:
@cdcapano I think for now I'll make this directly in the likelihood. I thought about adding to matched filter core, but I'm not sure the best strategy at the moment.